app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -793,6 +826,25 @@ public boolean matches(UserKeyRecord ukr) {
         CommCareApplication._().closeUserSession();
     }
 
+    public String getCurrentUserId() {
+        if (currentUserId != null) {
+            return currentUserId;
+        }
+        try {
+            currentUserId = this.getSession().getLoggedInUser().getUniqueId();

```

Comment:
Oh, I see. Hm, my guess is that the google analytics code behind reporting the event is going to take more time than doing this 3 part call (I don't think these methods have any ~body, it's just the stack setup).

If we're concerned about runtime I'd validate how much time this particular call takes against the rest of the report, if it's not pronounced I vote we kill it.